terraform/projects/app-ecs-instances/instance-user-data.tpl

```diff
@@ -1,8 +1,44 @@
 #!/bin/bash
+# Attach EBS volume to instance
+echo "[$(date '+%H:%M:%S %d-%m-%Y')] installing dependencies for volume attaching"
+sudo yum install -y aws-cli wget
 
-echo 'ECS_CLUSTER=${cluster_name}' >> /etc/ecs/ecs.config
+REGION="${region}"
+DEVICE="xvdf"
+VOLUME_ID="${volume_id}"
+
+echo "[$(date '+%H:%M:%S %d-%m-%Y')] finding current instance ID"
+INSTANCE_ID="`wget -q -O - http://169.254.169.254/latest/meta-data/instance-id`"
+
+echo "[$(date '+%H:%M:%S %d-%m-%Y')] attaching volume"
+aws ec2 attach-volume --volume-id $VOLUME_ID --instance-id $INSTANCE_ID --device /dev/$DEVICE --region $REGION
+
+# Waiting for volume to finish attaching
+x=0
+while [[ $x -lt 15 ]]; do
+  if ! [[ -e /dev/$DEVICE ]] ; then
+    sleep 1
+  else
+    break
+  fi
+  x=$((x+1))
+done
+
+# Format and mount volume
+if sudo file -s /dev/$DEVICE | grep -q "/dev/$DEVICE: data"; then
```

Comment:
do we need `sudo` in user data? I thought it ran as root anyway?

Reply:
we didn't, have changed